### PR TITLE
Declare out of disk when there is no free inodes

### PR DESCRIPTION
**Pull request by ronnielai**
#21546

### CODE REVIEW

File: pkg/kubelet/disk_manager.go [hunk 1]
```diff
@@ -102,11 +104,14 @@ func (dm *realDiskSpaceManager) isSpaceAvailable(fsType string, threshold int, f
 	if fsInfo.Available < 0 {
 		return true, fmt.Errorf("wrong available space for %q: %+v", fsType, fsInfo)
 	}
-
 	if fsInfo.Available < int64(threshold)*mb {
 		glog.Infof("Running out of space on disk for %q: available %d MB, threshold %d MB", fsType, fsInfo.Available/mb, threshold)
 		return false, nil
 	}
+	if fsInfo.InodesFree == 0 {
```
Comment (vishh) [inline pkg/kubelet/disk_manager.go]: Should we define a threshold for inodes as well that is > 0? By the time kubelet detects that there are inodes, the pods on the node are most likely stuck.

**Comment by vishh** [comment]
Just one comment. Otherwise LGTM

File: pkg/kubelet/disk_manager.go (hunk 1 above)
Comment (vishh) [inline pkg/kubelet/disk_manager.go]: My comment can be resolved in a subsequent PR too.

**Comment by derekwaynecarr** [comment]
This is a distinct condition from DiskPressure, correct?

https://github.com/kubernetes/kubernetes/pull/27199

(Which since I have been out appears to need a rebase next week)

**Pull request merged**